src/test/java/org/jenkinsci/plugins/github_branch_source/GithubSCMSourcePRsTest.java

```diff
@@ -0,0 +1,237 @@
+        githubApi.stubFor(
+                get(urlEqualTo("/repos/cloudbeers/yolo/pulls?state=open"))
+                        .willReturn(
+                                aResponse()
+                                        .withHeader("Content-Type", "application/json; charset=utf-8")
+                                        .withBodyFile("../PRs/_files/body-yolo-pulls-open-multiple-PRs.json")));
+        SCMHeadObserver mockSCMHeadObserver = Mockito.mock(SCMHeadObserver.class);
+        GitHubSCMSourceContext context = new GitHubSCMSourceContext(null, mockSCMHeadObserver);
+        context.wantOriginPRs(true);
+        GitHubSCMSourceRequest request = context.newRequest(new GitHubSCMSource("cloudbeers", "yolo", null, false), null);
+        Iterator<GHPullRequest> pullRequest = new GitHubSCMSource("cloudbeers", "yolo", null, false)
+                .new LazyPullRequests(request, repo).iterator();
+        // Expected: In the iterator will have 2 items in it
+        assertTrue(pullRequest.hasNext());
+        assertEquals(1, pullRequest.next().getId());
+        pullRequest.next();
+        assertTrue(pullRequest.hasNext());
+        assertEquals(2, pullRequest.next().getId());
+        pullRequest.next();
+        assertFalse(pullRequest.hasNext());
+    }
+
+    //Multiple PRs
+    @Test
+    public void testOpenMultiplePRsWithMasterAsOrigin() throws IOException {
+        // Situation: Hitting the Github API all the PRs and they are all Open but the master is the
+        githubApi.stubFor(
+                get(urlEqualTo("/repos/cloudbeers/yolo/pulls?state=open&head=cloudbeers%3Amaster"))
+                        .willReturn(
+                                aResponse()
+                                        .withHeader("Content-Type", "application/json; charset=utf-8")
+                                        .withBodyFile("../PRs/_files/body-yolo-pulls-open-multiple-PRs.json")));
+        SCMHeadObserver mockSCMHeadObserver = Mockito.mock(SCMHeadObserver.class);
+        GitHubSCMSourceContext context = new GitHubSCMSourceContext(null, mockSCMHeadObserver);
+        context.wantOriginPRs(true);
+        Set<SCMHead> masterSet = new HashSet<>();
+        SCMHead masterHead = new BranchSCMHead("master");
+        masterSet.add(masterHead);
+        GitHubSCMSourceContext contextSpy = Mockito.spy(context);
+        Mockito.when(contextSpy.observer().getIncludes()).thenReturn(masterSet);
+        GitHubSCMSourceRequest request = contextSpy.newRequest(new GitHubSCMSource("cloudbeers", "yolo", null, false), null);
+        Iterator<GHPullRequest> pullRequest = new GitHubSCMSource("cloudbeers", "yolo", null, false)
+                .new LazyPullRequests(request, repo).iterator();
+        // Expected: In the iterator will have 2 items in it
+        assertTrue(pullRequest.hasNext());
+        assertEquals(1, pullRequest.next().getId());
+        pullRequest.next();
```

Comment:
So, this is a bug you found right? Now that you have working tests, can you add a commit that fixes this?

Reply:
Added in the next commit